src/Fixer/FunctionNotation/VoidReturnFixer.php

```diff
@@ -0,0 +1,243 @@
+{
+    /**
+     * @internal
+     */
+    const VOID_RETURN_PATTERN = '/@return\s+void(?!\|)/';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Add void return type to functions with missing or empty return statements, but priority is given to `@return` annotations. Requires PHP >= 7.1.',
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\nfunction foo(\$a) {};",
+                    new VersionSpecification(70100)
+                ),
+            ],
+            'Rule is applied only in a PHP 7.1+ environment.',
+            'Modifies the signature of functions.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must run before ReturnTypeDeclarationFixer and PhpdocNoEmptyReturnFixer
+        return 15;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70100 && $tokens->isTokenKindFound(T_FUNCTION);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // These cause syntax errors.
+        static $blacklistFuncNames = [
+            [T_STRING, '__construct'],
+            [T_STRING, '__destruct'],
+            [T_STRING, '__clone'],
+        ];
+
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $funcName = $tokens->getNextMeaningfulToken($index);
+            if ($tokens[$funcName]->equalsAny($blacklistFuncNames, false)) {
+                continue;
+            }
+
+            $startIndex = $tokens->getNextTokenOfKind($index, ['{', ';']);
+            if (!$tokens[$startIndex]->equals('{')) {
+                // No function body defined, fallback to PHPDoc.
+                if ($this->hasVoidReturnAnnotation($tokens, $index)) {
+                    $this->fixFunctionDefinition($tokens, $startIndex);
+                }
+                continue;
+            }
+
+            if ($this->hasReturnAnnotation($tokens, $index)) {
+                continue;
+            }
+
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $startIndex);
+
+            if ($this->hasVoidReturn($tokens, $startIndex, $endIndex)) {
+                $this->fixFunctionDefinition($tokens, $startIndex);
+            }
+        }
+    }
+
+    /**
+     * Determine if there is a non-void return annotation in the function's PHPDoc comment.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the function token
+     *
+     * @return bool
+     */
+    private function hasReturnAnnotation(Tokens $tokens, $index)
```

Comment:
They are subtly different, `hasReturnAnnotation` is looking for any non-void return annotation while `hasVoidReturnAnnotation` is looking specifically for the existence of a void return annotation.  So `!$this->hasVoidReturnAnnotation(...)` would be a false positive when there are no PHPDocs at all.